tests/zfs-tests/cmd/zed_fd_spill-zedlet.c

```diff
@@ -28,7 +29,19 @@ int main(void) {
 
 	char buf[64];
 	sprintf(buf, "/tmp/zts-zed_fd_spill-logdir/%d", getppid());
-	dup2(creat(buf, 0644), STDOUT_FILENO);
+	int fd = creat(buf, 0644);
+	if (fd == -1) {
+		(void) fprintf(stderr, "creat(%s) failed: %s\n", buf,
+		    strerror(errno));
+		exit(EXIT_FAILURE);
+	}
+
+	if (dup2(fd, STDOUT_FILENO) != 0) {
+		close(fd);
+		(void) fprintf(stderr, "dup2(%s, STDOUT_FILENO) failed: %s\n",
+		    buf, strerror(errno));
+		exit(EXIT_FAILURE);
+	}
```

Comment:
After this change the `zed_fd_spill` test case is reliably failing.

Reply:
The check should be `== -1`. I do not know why I did `!= 0`. I will fix this shortly.

Reply:
A fix has been pushed.